src/chage.c

```diff
@@ -613,8 +612,8 @@ static void update_age (/*@null@*/const struct spwd *sp,
 		struct passwd pwent = *pw;
 
 		memzero(&spwent, sizeof(spwent));
-		spwent.sp_namp = xstrdup (pwent.pw_name);
-		spwent.sp_pwdp = xstrdup (pwent.pw_passwd);
+		spwent.sp_namp = pwent.pw_name;
+		spwent.sp_pwdp = pwent.pw_passwd;
 		spwent.sp_flag = SHADOW_SP_FLAG_UNSET;
 
```

Comment:
Thanks!  I was suspecting this duplication was indeed redundant, but didn't have the energy to follow through all the layers of functions.